components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -107,36 +115,58 @@ void ViewCounterService::BrandedWallpaperWillBeDisplayed(
   UpdateP3AValues();
 }
 
+#if BUILDFLAG(ENABLE_NTP_BACKGROUND_IMAGES)
 NTPBackgroundImagesData*
+ViewCounterService::GetCurrentWallpaperData() const {
+  return service_->GetBackgroundImagesData();
+}
+#endif
+
+NTPSponsoredImagesData*
 ViewCounterService::GetCurrentBrandedWallpaperData() const {
-  auto* sr_data = service_->GetBackgroundImagesData(true /* for_sr */);
+  auto* sr_data = service_->GetBrandedImagesData(true /* for_sr */);
   if (sr_data && IsSuperReferralWallpaperOptedIn())
     return sr_data;
 
-  return service_->GetBackgroundImagesData(false);
+  return service_->GetBrandedImagesData(false);
 }
 
 base::Value ViewCounterService::GetCurrentWallpaperForDisplay() const {
-  if (ShouldShowBrandedWallpaper()) {
+  bool shouldShowBrandedWallpaper = ShouldShowBrandedWallpaper();
+  if (shouldShowBrandedWallpaper) {
+    return GetCurrentBrandedWallpaper();
+  } else {
+#if BUILDFLAG(ENABLE_NTP_BACKGROUND_IMAGES)
     return GetCurrentWallpaper();
```

Comment:
`GetCurrentWallpaperForDisplay()` purpose is to take in to account what data is available, what the page count number is **and** what the user preferences are to be the 1 function which will provide _the_ wallpaper to display. So, here we have to take in to account whether the user has background images enabled via the `kNewTabPageShowBackgroundImage` pref. Unless android does not have a preference to turn off background image?